keychain/src/types.rs

```diff
@@ -468,7 +468,19 @@ pub trait Keychain: Sync + Send + Clone {
 	fn derive_key(&self, amount: u64, id: &Identifier) -> Result<SecretKey, Error>;
 	fn commit(&self, amount: u64, id: &Identifier) -> Result<Commitment, Error>;
 	fn blind_sum(&self, blind_sum: &BlindSum) -> Result<BlindingFactor, Error>;
-	fn create_nonce(&self, commit: &Commitment) -> Result<SecretKey, Error>;
+
+	/// Create a BP nonce that will allow to rewind the derivation path and flags
+	fn create_rewind_nonce(&self, commit: &Commitment, legacy: bool) -> Result<SecretKey, Error>;
```

Comment:
When we redesign code we're really just replacing today's legacy code with tomorrow's.  Let's make it easier on our future selves!  Consider substituting a version byte for the legacy bool.